Letterbook.Adapter.ActivityPub/Client.cs

```diff
@@ -110,22 +110,34 @@ private HttpRequestMessage SignedRequest(HttpMethod method, Uri uri, ASType docu
 	}
 
 	private HttpRequestMessage SignedRequest(HttpMethod method, Uri uri)
+	{
+		return SignedRequest(method, uri, _actor?.Keys, _actor?.FediId);
+	}
+
+	private HttpRequestMessage SignedRequest(HttpMethod method, Uri uri, IEnumerable<SigningKey>? keys, Uri? actorId)
 	{
 		var message = new HttpRequestMessage(method, uri);
-		if (_profile == null) return message;
+		if (keys is null || actorId is null) return message;
 
 		var httpRequestOptionsKey = new HttpRequestOptionsKey<IEnumerable<SigningKey>>(IClientSigner.SigningKeysOptionsId);
 		var httpRequestOptionsProfile = new HttpRequestOptionsKey<Uri>(IClientSigner.ProfileOptionsId);
-		message.Options.Set(httpRequestOptionsKey, _profile.Keys);
-		message.Options.Set(httpRequestOptionsProfile, _profile.FediId);
+		message.Options.Set(httpRequestOptionsKey, keys);
+		message.Options.Set(httpRequestOptionsProfile, actorId);
 
 		return message;
 	}
 
+
 	public async Task<ClientResponse<FollowState>> SendFollow(Uri inbox, Models.Profile target)
 	{
-		var doc = _document.Follow(_profile!, target);
-		doc.Id = $"{_profile!.FediId}#follow/{target.FediId}";
+		if (_actor is not Models.Profile profileActor)
+		{
+			// Not sure if this is the right thing to do. Can instances follow other actors?
```

Comment:
BTW, since I decided against using the instance actor for client fetches (for now), and opted for an overload of Fetch that allows specifying the signing key instead, there's also the option to roll this and related changes back.